### PR TITLE
Update reason_cache.go, Get method operate lru cache not threadsafe

### PR DESCRIPTION
**Pull request by 249043822**
The reason_cache wrapped lru cache , lru cache modies linked list even for a get, should use WLock for both read and write

### CODE REVIEW

**Comment by Random-Liu** [comment]
ok to test

**Comment by Random-Liu** [comment]
@249043822 Totally make sense. Thanks for the fix! :)

**Comment by Random-Liu** [comment]
@k8s-bot ok to test

**Comment by 249043822** [comment]
@Random-Liu @vishh , who can give lgtm and ok-to-merge? thanks

**Comment by Random-Liu** [comment]
Mark 1.3 because this fixed a potential data race and code change is minor.

File: pkg/kubelet/reason_cache.go [hunk 1]
```diff
@@ -93,8 +93,8 @@ func (c *ReasonCache) Remove(uid types.UID, name string) {
 // whether an error reason is found in the cache. If no error reason is found, empty string will
 // be returned for error reason and error message.
 func (c *ReasonCache) Get(uid types.UID, name string) (error, string, bool) {
-	c.lock.RLock()
-	defer c.lock.RUnlock()
+	c.lock.Lock()
+	defer c.lock.Unlock()
```
Comment (dims) [inline pkg/kubelet/reason_cache.go]: good catch. Same fix was applied to pkg/controller/lookup_cache.go as well a few days ago.

https://github.com/kubernetes/kubernetes/pull/26758/commits/9a0ff5a9e8a8fe4fb6f6ac0c8b8eb8597c19ff40

**Comment by 249043822** [comment]
Hi @Random-Liu , could you help what happened about Jenkins GCE e2e — No test results found.
Thanks

**Comment by Random-Liu** [comment]
@k8s-bot test this issue #27214

**Comment by lavalamp** [comment]
@Random-Liu Would you expect this to make our unit tests a lot slower? They got like 16 minutes slower after this merged.

**Comment by Random-Liu** [comment]
@lavalamp Really?? I don't think this is tested so much, I'll check if there is something wrong.

**Comment by Random-Liu** [comment]
I checked kubernetes-test-go 12984(54min) and 12985(with this PR, 1hr8min).
The total build (including code verification) time is increased 24min ==> 29min, increased by 5 min.
The total test time including unit test and integration test is almost the same: 2479.17s ==> 2450.34s.
So there should be something else changed, making test-go much slower.

**Comment by Random-Liu** [comment]
@lavalamp I checked kubernetes-test-go 12984(54min) and 12985(with this PR, 1hr8min).
The total build (including code verification) time is increased 24min ==> 29min, increased by 5min.
The unit test time (from unit test start to finish) is increased 13 min ==> 23 min, increased by 10min.
The integration test time is almost not changed.

But the wired thing is that if we sum up the unit test time reported by go, it is not changed much: 1783.3s ==> 1762.65s.

**Comment by lavalamp** [comment]
Thanks Lantao. I think this PR is probably not the culprit.

**Pull request merged**